test/index.spec.js

```diff
@@ -87,4 +88,93 @@ describe('pull-length-prefixed', () => {
       })
     )
   })
+
+  const sizes = [1, 2, 4, 6, 10, 100, 1000]
+
+  sizes.forEach((size) => {
+    it(`split packages to blocks: ${size}`, (done) => {
+      const longBuffer = new Buffer(size * 10)
+      longBuffer.fill('a')
+
+      const input = [
+        new Buffer('hello '),
+        new Buffer('world'),
+        longBuffer
+      ]
+
+      pull(
+        pull.values(input),
+        lp.encode(),
+        block(size, {nopad: true}),
+        lp.decode(),
+        pull.collect((err, res) => {
+          if (err) throw err
+
+          expect(
+            res
+          ).to.be.eql([
+            new Buffer('hello '),
+            new Buffer('world'),
+            longBuffer
+          ])
+          done()
+        })
+      )
+    })
+  })
+
+  describe('back pressure', () => {
+    let input = []
+
+    before(() => {
+      for (let j = 0; j < 200; j++) {
+        const a = []
+        for (let i = 0; i < 200; i++) {
+          a[i] = String(i)
+        }
+
+        input.push(new Buffer(a.join('')))
+      }
+    })
+
+    it('encode - slow in - fast out', (done) => {
+      pull(
+        pull.values(input),
+        delay(10),
```

Comment:
I was just trying to simulate different scenarios. If the connection is fast it is like in the other cases, where encode is directly piped into decode.